src/compiler/checker.ts

```diff
@@ -24494,12 +24496,12 @@ namespace ts {
             return prop !== undefined && prop.valueDeclaration && hasModifier(prop.valueDeclaration, ModifierFlags.Static);
         }
 
-        function getSuggestedSymbolForNonexistentProperty(name: Identifier | PrivateIdentifier | string, containingType: Type): Symbol | undefined {
-            return getSpellingSuggestionForName(isString(name) ? name : idText(name), getPropertiesOfType(containingType), SymbolFlags.Value);
+        function getSuggestedSymbolForNonexistentProperty(name: Identifier | PrivateIdentifier | string, containingType: Type, isJsxAttr: boolean): Symbol | undefined {

```

Comment:
I would fork here and write

```ts
function getSuggestedSymbolForNonexistentJSXAttribute(name: string, containingType: Type) {
  const properties = getPropertiesOfType(containingType)
  if (name === "for" && contains(properties, name)) return "htmlFor"
  if (name === "class" && contains(properties, name)) return "className"
  else return getSpellingSuggestionForName(name, properties)
}
```

Then just call this directly in the two places where you currently pass `/*isJsx*/ true`.